ironic/resource_ironic_deployment.go

```diff
@@ -94,7 +95,19 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
 	// Set instance info
 	instanceInfo := d.Get("instance_info").(map[string]interface{})
 	if instanceInfo != nil {
-		_, err := UpdateNode(client, d.Get("node_uuid").(string), nodes.UpdateOpts{
+		instanceInfoCapabilities, found := instanceInfo["capabilities"]
+		capabilities := make(map[string]string)
+		nodeUUID := d.Get("node_uuid").(string)
+		if found {
+			for _, e := range strings.Split(instanceInfoCapabilities.(string), ",") {
+				parts := strings.Split(e, ":")
+				if len(parts) == 2 {
+					capabilities[parts[0]] = parts[1]
+				}
```

Comment:
What do you think about logging or return an error if len isn't 2?

Reply:
sure!